src/FlipMove.js

```diff
@@ -233,6 +231,23 @@ class FlipMove extends Component {
         domNode.style.left  = leavingBoundingBox.left - cleanedComputed['margin-left'] + 'px';
         domNode.style.right = leavingBoundingBox.right - cleanedComputed['margin-right'] + 'px';
       });
+
+      if ( this.props.maintainContainerHeight ) {
+        // We need to find the height of the container *without* the padding
+        // element. Since it's possible that the padding element might already
+        // be present, we first set its height to 0. This allows the container to
+        // collapse down to the size of just its content.
+        this.heightPlaceholder.style.height = 0;
+        const contentHeight = this.props.getPosition(this.parentElement).height;
+
+        // Find the distance by which the container would be collapsed by elements
+        // leaving. We compare the temporarily available `collapsedHeight` with
+        // the previously cached container height.
+        const collapseHeight = this.parentBox.height - contentHeight;
+
+        // Update the padding element's height.
+        this.heightPlaceholder.style.height = Math.max(0, collapseHeight) + 'px';
```

Comment:
So it's more verbose, but I think it might be clearer if it was something like:

```js
// First two lines unchanged
this.heightPlaceholder.style.height = 0;
const contentHeight = this.props.getPosition(this.parentElement).height;

const hasContentShrunk = contentHeight < this.parentBox.height;

if ( hasContentShrunk ) {
  const changeInHeight = this.parentBox.height - contentHeight;
  this.heightPlaceholder.style.height = `${changeInHeight}px`;
} else {
  this.heightPlaceholder.style.height = 0;
}
```

I think it's clearer because it's obvious that we're only setting the placeholder's height to the diff when the content has shrunk.

(Also, thanks for showing me that `Math.max` trick; really nice built-in alternative to lodash's `clamp`)